clusterloader2/pkg/measurement/common/pod_startup_latency.go

```diff
@@ -0,0 +1,302 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	podStartupLatencyMetricName = "PodStartupLatency"
+	syncTimeout                 = time.Minute
+	resyncPeriod                = time.Second
+)
+
+func init() {
+	measurement.Register(podStartupLatencyMetricName, createPodStartupLatencyMeasurement)
+}
+
+func createPodStartupLatencyMeasurement() measurement.Measurement {
+	return &podStartupLatencyMeasurement{}
+}
+
+type podStartupLatencyMeasurement struct {
+	namespace     string
+	labelSelector string
+	fieldSelector string
+	informer      cache.SharedInformer
+	stopCh        chan struct{}
+	createTimes   map[string]map[string]metav1.Time
```

Comment:
IIUC this is map from namespace -> createTimes.

Do you want to allow for the same names across namespaces?
Then much easier would be to make the key of the form "namespace/name" [it's done in many places already, e.g. it's what informer cache is also doing].

MetaNamespaceKey (or sth like) that in tools/cache in client-go is what you're looking for

Reply:
Changed to "namespace/name".